sklearn/linear_model/stochastic_gradient.py

```diff
@@ -102,6 +102,9 @@ def _validate_params(self):
         if self.learning_rate in ("constant", "invscaling"):
             if self.eta0 <= 0.0:
                 raise ValueError("eta0 must be > 0")
+        if self.learning_rate == "optimal":
+            if self.alpha == 0:
+                raise ValueError("alpha must be > 0")
```

Comment:
Could you make a more explicit message, explaining that it is in the case of `learning_rate=='optimal'`?